src/test/mir-opt/let-chains.rs

```diff
@@ -0,0 +1,94 @@
+// ignore-wasm32-bare compiled with panic=abort by default
+// compile-flags: -Z mir-opt-level=0
+
+// See `mir_let_chains_drop_order.rs` and `mir_drop_order` for more information
+
+#![feature(let_chains)]
+
+use std::cell::RefCell;
+use std::panic;
+
+pub struct DropLogger<'a, T> {
+    extra: T,
+    id: usize,
+    log: &'a panic::AssertUnwindSafe<RefCell<Vec<usize>>>
+}
+
+impl<'a, T> Drop for DropLogger<'a, T> {
+    fn drop(&mut self) {
+        self.log.0.borrow_mut().push(self.id);
+    }
+}
+
+struct InjectedFailure;
+
+// EMIT_MIR let_chains.main.ElaborateDrops.before.mir
```

Comment:
Why is this a MIR opt test? It has far too much code to have people able to check the output MIR.

Reply:
Not sure why but I removed anyway. The similar UI test is still present

cc @Aaron1011